test/bench/palindrome.mo

```diff
@@ -0,0 +1,62 @@
+// The 2 palindrome implementations from
+// "There and Back Again", by Olivier Danvy and Mayer Goldberg
+//
+import { error; performanceCounter; rts_heap_size; debugPrint } = "mo:⛔";
+
+actor Palindrome {
+    type List<A> = ?(A, List<A>);
+
+    func cps(xs : List<Char>) : Bool {
+        func walk(xs1 : List<Char>, xs2 : List<Char>, k : List<Char> -> Bool) : Bool =
+            switch (xs1, xs2) {
+                 case (_, null) { k xs1 };
+                 case (?(_, xs1), ?(_, null)) { k xs1 };
+                 case (?(x, xs1), ?(_, ?(_, xs2))) {
+                     walk (xs1, xs2, func (?(y, ys)) = x == y and k ys)
+                 }
+             };
+        walk (xs, xs, func null = true)
```

Comment:
`func null = true` will trap on non-empty list - don't you want to return false instead?
`func l { l == null;}` might work better.

Reply:
No, this works and is as close to the original as possible.